src/isomorphic/modern/class/ReactBaseClasses.js

```diff
@@ -27,6 +27,9 @@ function ReactComponent(props, context, updater) {
   this.refs = emptyObject;
   // We initialize the default updater but the real one gets injected by the
   // renderer.
+  if (this.updater && "development" !== 'production') {
+      "development" !== 'production' ? warning(this instanceof Constructor, 'This object has a property called updater which will be overwritten by React default updater') : void 0;
```

Comment:
What are you trying to do here?

The problem we're trying to solve is that if `instance.updater` is mutated, it will break `setState` (and `replaceState`). "Updater" is a common enough word that someone might do that accidentally without realizing that it's a private React method:

```js
class MyComponent extends React.Component{
  constructor() {
    // This will break setState :(
    this.updater = someCustomValue
  }
  /* ... */
}
```

I think we should either warn whenever `updater` is mutated, or obscure the name.

Since it's possible someone is relying on `updater`, let's go with the first option and print a warning. Something like

> ComponentName: The instance variable `updater` is an internal React method. Mutating it is not supported and it may be changed or removed in a future release.